### PR TITLE
TSK-7092: files v2 adjustments

### PR DESCRIPTION
**Pull request by abramov-blink**

### CODE REVIEW

File: plugin/config/aws.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	"fmt"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// SetAwsEnv sets the env vars that are read by the AWS SDK to acquire credentials.
+// It returns a function which unsets these env vars
+func SetAwsEnv(accessKeyId, accessSecretKey, sessionToken string) (func(), error) {
```
Comment (itsofirblink) [inline plugin/config/aws.go:13]: i think that's a big no no, don't set env variables with credentials on the root user.
You can create object `credentials` to pass to all AWS SDK functions

File: plugin/files/aws_provider.go [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+			downloadObjects[i] = s3manager.BatchDownloadObject{
+				Object: &s3.GetObjectInput{
+					Bucket: aws.String(p.bucket),
+					Key:    aws.String(filepath.Join(p.rootDir, request.FileName)),
+				},
+				Writer: writerAt,
+			}
+		}
+	}
```
Comment (itsofirblink) [inline plugin/files/aws_provider.go:94]: i don't understand why won't you just use the `else` part with 
```golang
downloader.Concurrency = numRequests
```
or something like that, that's basically the same

File: plugin/files/aws_provider.go [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+	unsetAwsEnv, err := config.SetAwsEnv(p.accessKeyId, p.accessSecretKey, p.sessionToken)
+	if err != nil {
+		return err
+	}
+
+	defer unsetAwsEnv()
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(p.region),
+	})
```
Comment (itsofirblink) [inline plugin/files/aws_provider.go:111]: extract config to a variable

File: plugin/files/aws_provider.go [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+	}
+
+	downloader, downloadObjects, err := p.getDownloaderAndObjectsToDownload(sess, requests)
+	if err != nil {
+		return err
+	}
+
+	if err = downloader.DownloadWithIterator(ctx, &s3manager.DownloadObjectsIterator{
+		Objects: downloadObjects,
+	}); err != nil {
```
Comment (itsofirblink) [inline plugin/files/aws_provider.go:123]: extract iterator to a variable

File: plugin/files/aws_provider.go [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+				Bucket: aws.String(p.bucket),
+				Key:    aws.String(filepath.Join(p.rootDir, request.FileName)),
+				Body:   request.Reader,
+			},
+		}
+	}
+
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(p.region),
+	})
```
Comment (itsofirblink) [inline plugin/files/aws_provider.go:154]: extract config to variable

File: plugin/files/aws_provider.go [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+	if err != nil {
+		return fmt.Errorf("error intializing aws session before uploading to s3: %v", err)
+	}
+
+	if err = s3manager.NewUploader(sess).UploadWithIterator(ctx, &s3manager.UploadObjectsIterator{
+		Objects: uploadObjects,
```
Comment (itsofirblink) [inline plugin/files/aws_provider.go:160]: extract iterator to a variable

**Review by itsofirblink — CHANGES_REQUESTED** [review]

File: plugin/config/aws.go (hunk 1 above)
Comment (abramov-blink) [reply plugin/config/aws.go]: I agree. Done

File: plugin/files/aws_provider.go (hunk 2 above)
Comment (abramov-blink) [reply plugin/files/aws_provider.go]: Done (something like that.. like we talked about :) )

File: plugin/files/aws_provider.go (hunk 3 above)
Comment (abramov-blink) [reply plugin/files/aws_provider.go]: Done

File: plugin/files/aws_provider.go (hunk 4 above)
Comment (abramov-blink) [reply plugin/files/aws_provider.go]: Done

File: plugin/files/aws_provider.go (hunk 5 above)
Comment (abramov-blink) [reply plugin/files/aws_provider.go]: Done

File: plugin/files/aws_provider.go [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+	}
+
+	awsConfig := &aws.Config{
+		Credentials: awsCreds.NewStaticCredentials(p.accessKeyId, p.accessSecretKey, p.sessionToken),
+		Region:      aws.String(p.region),
+	}
+	sess, err := session.NewSession(awsConfig)
+	if err != nil {
+		return fmt.Errorf("error intializing aws session before uploading to s3: %v", err)
+	}
```
Comment (itsofirblink) [inline plugin/files/aws_provider.go:167]: extract to function `getSession`

File: plugin/files/aws_provider.go [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+
+	awsConfig := &aws.Config{
+		Credentials: awsCreds.NewStaticCredentials(p.accessKeyId, p.accessSecretKey, p.sessionToken),
+		Region:      aws.String(p.region),
+	}
+	sess, err := session.NewSession(awsConfig)
+	if err != nil {
+		return fmt.Errorf("error intializing aws session before downloading from s3: %v", err)
+	}
```
Comment (itsofirblink) [inline plugin/files/aws_provider.go:127]: use `getSession`

File: plugin/files/aws_provider.go [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+
+func (p *awsProvider) CheckIfFileExists(fileName string) (bool, error) {
+	awsConfig := &aws.Config{
+		Credentials: awsCreds.NewStaticCredentials(p.accessKeyId, p.accessSecretKey, p.sessionToken),
+		Region:      aws.String(p.region),
+	}
+	sess, err := session.NewSession(awsConfig)
+	if err != nil {
+		return false, fmt.Errorf("error intializing aws session before checking file existence in s3: %v", err)
+	}
```
Comment (itsofirblink) [inline plugin/files/aws_provider.go:189]: use `getSession`

File: plugin/files/aws_provider.go [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+
+	if _, err = s3.New(sess).HeadObject(&s3.HeadObjectInput{
+		Bucket: aws.String(p.bucket),
+		Key:    aws.String(filepath.Join(p.rootDir, fileName)),
+	}); err != nil {
```
Comment (itsofirblink) [inline plugin/files/aws_provider.go:194]: extract `HeadObjectInput` to var

File: plugin/files/files.go [hunk 11]
```diff
@@ -42,163 +45,313 @@ func Get(ctx *plugin.ActionContext) (map[string]*File, error) {
...
-		return err
+		return "", 0, err
 	}
 
-	for _, file := range files {
-		if err = fromContext(file); err != nil {
-			return err
-		}
+	if !regexp.MustCompile(FileNameRegex).MatchString(fileName) {
+		return "", 0, fmt.Errorf("filename '%s' is invalid. Should only contain digits/numbers/'-'/'.'", fileName)
```
Comment (itsofirblink) [inline plugin/files/files.go:107]: you can print out the regex

File: plugin/action_context.go [hunk 12]
```diff
@@ -73,6 +92,42 @@ func resolveInnerItems(key string, createKeys bool, innerContext map[string]inte
...
 }
 
+func (ctx *ActionContext) populateFilesMetadataFields() {
+	filesMetadata, exists := ctx.internalContext[FilesMetadataCtxKey]
+	if !exists {
+		return
+	}
+
+	filesMetadataMap, ok := filesMetadata.(map[string]any)
+	if ok {
```
Comment (itsofirblink) [inline plugin/action_context.go:102]: revese, `!ok` -> `return`

File: plugin/files/aws_provider.go (hunk 7 above)
Comment (abramov-blink) [reply plugin/files/aws_provider.go]: Done

File: plugin/files/aws_provider.go (hunk 8 above)
Comment (abramov-blink) [reply plugin/files/aws_provider.go]: Done

File: plugin/files/aws_provider.go (hunk 9 above)
Comment (abramov-blink) [reply plugin/files/aws_provider.go]: Done

File: plugin/files/aws_provider.go (hunk 10 above)
Comment (abramov-blink) [reply plugin/files/aws_provider.go]: Done

File: plugin/files/files.go (hunk 11 above)
Comment (abramov-blink) [reply plugin/files/files.go]: Done

File: plugin/action_context.go (hunk 12 above)
Comment (abramov-blink) [reply plugin/action_context.go]: Done

**Review by itsofirblink — APPROVED** [review]

**Pull request merged**